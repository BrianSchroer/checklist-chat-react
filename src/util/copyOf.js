export default original => Object.assign({}, original);
