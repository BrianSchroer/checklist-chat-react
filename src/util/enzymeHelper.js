const enzymeHelper = class {

    static nestedFind(wrapper, selector) {
        const selectors = selector.split('>').map(node => node.trim());
        let parent = wrapper;
        let found = [];

        for (let i = 0; i < selectors.length; i++) {
            found = parent.find(selectors[i]);
            if (!found.length) {
                break;
            }
        }

        return found;
    }

    static findSingle(wrapper, selector) {
        const findArray = enzymeHelper.nestedFind(wrapper, selector);

        if (findArray.length != 1) {
            throw(`expected 1 find for selector "${selector}", but found ${findArray.length}.`);
        }

        return findArray.first();
    }
};

export default enzymeHelper;
