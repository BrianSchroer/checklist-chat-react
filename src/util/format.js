const localeTimeStringOptions = {
  hour: 'numeric',
  minute: '2-digit'
};

export default class {
  static time(input = '') {
    const isoDateFormatRegex = /\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}([+-]\d{2}:\d{2})?/g;
    let formatted = input;

    if (input && input.match(isoDateFormatRegex)) {
      const milliseconds = Date.parse(input);
      if (milliseconds) {
        const dt = new Date(milliseconds);
        const localeString = dt.toLocaleTimeString(
          'en-US',
          localeTimeStringOptions
        );
        formatted = localeString.toLowerCase();
      }
    }

    return formatted;
  }

  static isoDateTimeString(input) {
    let formatted = input;

    if (input) {
      const milliseconds = Date.parse(input);
      if (milliseconds) {
        formatted = new Date(milliseconds).toISOString();
      }
    }

    return formatted;
  }
}
