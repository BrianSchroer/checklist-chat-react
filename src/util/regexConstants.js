export const CSS_PATH = /(\.css)$/;
export const EOT_PATH = /\.eot(\?v=\d+\.\d+\.\d+)?$/;
export const JS_PATH = /\.js$/;
export const SVG_PATH = /\.svg(\?v=\d+\.\d+\.\d+)?$/;
export const TTF_PATH = /\.ttf(\?v=\d+\.\d+\.\d+)?$/;
export const WOFF_OR_WOFF2_PATH = /\.(woff|woff2)(\?v=\d+\.\d+\.\d+)?$/;
