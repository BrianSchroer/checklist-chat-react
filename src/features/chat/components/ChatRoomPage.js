import React, {PropTypes} from 'react';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import * as uiHelpers from '../../../util/uiHelpers';
import {requestChecklistItemModalDialog, requestRoomInfoModalDialog} from '../../../app/modalDialogDucks';
import {joinChat} from '../../../features/room/roomDucks';
import RoomInfo from '../../../features/room/components/RoomInfo';
import ChatMessageList from './ChatMessageList';
import ChatButtons from './ChatButtons';  // eslint-disable-line import/no-named-as-default
import Checklist from '../../../features/checklist/components/Checklist';
import {ChecklistButtons} from '../../../features/checklist/components/ChecklistButtons';

export class ChatRoomPage extends React.Component {
    constructor(props, context) {
        super(props, context);

        this.state = {
            room: Object.assign({}, props.room)
        };

        this.handleRoomInfoEditRequest = this.handleRoomInfoEditRequest.bind(this);
        this.handleChecklistItemEditRequest = this.handleChecklistItemEditRequest.bind(this);
    }

    componentWillMount() {
        const {actions, userId, routeParams} = this.props;

        actions.joinChat(routeParams.id, userId);
    }

    componentWillReceiveProps(nextProps) {
        let shouldScrollMessages = false;

        if (nextProps.chatMessages) {
            if (this.props && this.props.chatMessages) {
                shouldScrollMessages =
                    (nextProps.chatMessages.length != this.props.chatMessages.length);
            }
        }

        this.setState({shouldScrollMessages});
    }

    componentDidUpdate() {
        if (this.state.shouldScrollMessages) {
            uiHelpers.afterRenderIsComplete(() => {
                uiHelpers.scrollToBottom('chatMessageList');
            });
        }
    }

    handleRoomInfoEditRequest(event) {
        event.preventDefault();
        this.props.actions.requestRoomInfoModalDialog(this.props.room.id);
    }

    handleChecklistItemAddRequest(event) {
        event.preventDefault;
        this.props.actions.requestChecklistItemModalDialog(this.props.room.id);
    }

    handleChecklistItemEditRequest(event, roomId, sequenceNumber) {
        event.preventDefault();
        this.props.actions.requestChecklistItemModalDialog(roomId, sequenceNumber);
    }

    render() {
        const room = this.props.room;

        return (
            <div className="chat-room-page">
                <div className="chat-room-chat-column">
                    <div className="chat-room-room-info">
                        {room && <RoomInfo room={room} onEditRequest={this.handleRoomInfoEditRequest} />}                    </div>
                    <ChatMessageList chatMessages={this.props.chatMessages} />
                    <div className="chat-room-buttons">
                        <ChatButtons/>
                    </div>
                </div>
                <div className="chat-room-checklist-column">
                    <Checklist checklistItems={this.props.checklistItems}
                        OnEditRequest={this.handleChecklistItemEditRequest}/>
                    <div className="chat-room-buttons">
                        <ChecklistButtons/>
                    </div>
                </div>
            </div>
        );
    }
}

ChatRoomPage.propTypes = {
    routeParams: PropTypes.object.isRequired,
    userId: PropTypes.string.isRequired,
    room: PropTypes.object,
    checklistItems: PropTypes.arrayOf(PropTypes.object).isRequired,
    chatMessages: PropTypes.arrayOf(PropTypes.object).isRequired,
    actions: PropTypes.object.isRequired
};

function mapStateToProps(state) {
    const room = state.rooms.find(room => room.id == state.roomId);
    const userId = state.userId;
    const checklistItems = state.checklistItems;
    const chatMessages = state.chatMessages;

    return {userId, room, checklistItems, chatMessages};
}

const mapDispatchToProps = (dispatch) => ({
    actions: bindActionCreators({
        requestChecklistItemModalDialog, requestRoomInfoModalDialog,
        joinChat
    }, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(ChatRoomPage);
