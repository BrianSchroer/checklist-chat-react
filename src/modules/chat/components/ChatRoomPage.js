import React, {PropTypes} from 'react';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import {uiHelper} from '../../../util';
import * as modalDialogActions from '../../../app/modalDialogDucks';
import {joinChat} from '../../../modules/room/roomDucks';
import {RoomInfo} from '../../../modules/room';
import ChatMessageList from './ChatMessageList';
import ChatButtons from './ChatButtons';  // eslint-disable-line import/no-named-as-default
import Checklist from '../../../modules/checklist/components/Checklist';
import ChecklistButtons from '../../../modules/checklist/components/ChecklistButtons';

export class ChatRoomPage extends React.Component {

    static get propTypes() {
        return {
            routeParams: PropTypes.object.isRequired,
            userId: PropTypes.string.isRequired,
            room: PropTypes.object,
            checklistItems: PropTypes.arrayOf(PropTypes.object).isRequired,
            chatMessages: PropTypes.arrayOf(PropTypes.object).isRequired,
            actions: PropTypes.object.isRequired
        };
    }

    constructor(props, context) {
        super(props, context);

        this.state = {
            room: Object.assign({}, props.room)
        };

        this.handleRoomInfoEditRequest = this.handleRoomInfoEditRequest.bind(this);
        this.handleChatMessageAddRequest = this.handleChatMessageAddRequest.bind(this);
        this.handleChatParticipantsRequest = this.handleChatParticipantsRequest.bind(this);
        this.handleChecklistItemAddRequest = this.handleChecklistItemAddRequest.bind(this);
        this.handleChecklistItemEditRequest = this.handleChecklistItemEditRequest.bind(this);
    }

    componentWillMount() {
        const {actions, userId, routeParams} = this.props;

        actions.joinChat(routeParams.id, userId);
    }

    componentWillReceiveProps(nextProps) {
        let shouldScrollMessages = false;

        if (nextProps.chatMessages) {
            if (this.props && this.props.chatMessages) {
                shouldScrollMessages =
                    (nextProps.chatMessages.length != this.props.chatMessages.length);
            }
        }

        this.setState({shouldScrollMessages});
    }

    componentDidUpdate() {
        if (this.state.shouldScrollMessages) {
            uiHelper.afterRenderIsComplete(() => {
                uiHelper.scrollToBottom('chatMessageList');
            });
        }
    }

    handleRoomInfoEditRequest(event) {
        event.preventDefault();
        this.props.actions.requestRoomInfoModalDialog(this.props.room.id);
    }

    handleChatMessageAddRequest(event) {
        event.preventDefault();
        this.props.actions.requestChatMessageModalDialog();
    }

    handleChatParticipantsRequest(event) {
        event.preventDefault();
        this.props.actions.requestChatParticipantsModalDialog();
    }

    handleChecklistItemAddRequest(event) {
        event.preventDefault;
        this.props.actions.requestChecklistItemModalDialog(this.props.room.id);
    }

    handleChecklistItemEditRequest(event, roomId, sequenceNumber) {
        event.preventDefault();
        this.props.actions.requestChecklistItemModalDialog(roomId, sequenceNumber);
    }

    render() {
        const {room, userId, chatMessages} = this.props;

        return (
            <div className="chat-room-page">
                <div className="chat-room-chat-column">
                    <div className="chat-room-room-info">
                        {room && <RoomInfo room={room} onEditRequest={this.handleRoomInfoEditRequest} />}                    </div>
                    <ChatMessageList userId={userId} chatMessages={chatMessages} />
                    <div className="chat-room-buttons">
                        <ChatButtons
                            onChatMessageAddRequest={this.handleChatMessageAddRequest}
                            onChatParticipantsRequest={this.handleChatParticipantsRequest}
                        />
                    </div>
                </div>
                <div className="chat-room-checklist-column">
                    <Checklist checklistItems={this.props.checklistItems}
                        onEditRequest={this.handleChecklistItemEditRequest}/>
                    <div className="chat-room-buttons">
                        <ChecklistButtons onChecklistItemAddRequest={this.handleChecklistItemAddRequest} />
                    </div>
                </div>
            </div>
        );
    }
}

export function mapStateToProps(state) {
    const room = state.rooms.find(room => room.id == state.roomId);
    const userId = state.userId;
    const checklistItems = state.checklistItems;
    const chatMessages = state.chatMessages;

    return {userId, room, checklistItems, chatMessages};
}

const mapDispatchToProps = (dispatch) => ({
    actions: bindActionCreators({
        requestChatMessageModalDialog: modalDialogActions.requestChatMessageModalDialog,
        requestChatParticipantsModalDialog: modalDialogActions.requestChatParticipantsModalDialog,
        requestChecklistItemModalDialog: modalDialogActions.requestChecklistItemModalDialog,
        requestRoomInfoModalDialog: modalDialogActions.requestRoomInfoModalDialog,
        joinChat
    }, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(ChatRoomPage);
