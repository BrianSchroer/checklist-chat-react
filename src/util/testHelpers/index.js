import {shallow, mount} from 'enzyme';
import enzymeHelper from './enzymeHelper';

export {
    shallow as shallow,
    mount as mount,
    enzymeHelper as enzymeHelper
};
