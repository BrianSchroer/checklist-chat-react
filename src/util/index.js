import format from './format';
import uiHelper from './uiHelper';
import validationHelper from './validationHelper';

export {
    format as format,
    uiHelper as uiHelper,
    validationHelper as validationHelper
};
