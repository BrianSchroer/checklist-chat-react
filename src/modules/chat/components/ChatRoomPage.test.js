import React from 'react';
import {shallow, enzymeHelper} from '../../../util/testHelpers';
import {ChatRoomPage} from './ChatRoomPage';

const defaultProps = {
    routeParams: {},
    userId: 'TestUserId',
    room: {id: 123},
    checklistItems: [],
    chatMessages: [],
    actions: {
        joinChat: () => {}
    }
};

function render(propOverrides = {}) {
    const props = Object.assign({}, defaultProps, propOverrides);
    return shallow(<ChatRoomPage {...props} />);
}

describe('ChatRoomPage', () => {
    it('should render RoomInfo', () => {
        const roomInfo = enzymeHelper.findSingle(render(),
            'div.chat-room-page > div.chat-room-chat-column > div.chat-room-room-info > RoomInfo');
        expect(roomInfo.props().room).toBe(defaultProps.room);
    });

    it('should render ChatButtons', () => {
        enzymeHelper.findSingle(render(),
            'div.chat-room-page > div.chat-room-chat-column > div.chat-room-buttons > ChatButtons');
    });

    it('should render Checklist', () => {
        enzymeHelper.findSingle(render(),
            'div.chat-room-page > div.chat-room-checklist-column > Checklist');
    });

    it('should render ChecklistButtons', () => {
        enzymeHelper.findSingle(render(),
            'div.chat-room-page > div.chat-room-checklist-column > div.chat-room-buttons > ChecklistButtons');
    });
});
