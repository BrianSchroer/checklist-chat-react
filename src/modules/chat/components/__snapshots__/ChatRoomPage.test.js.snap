// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatRoomPage should render correctly 1`] = `
<div
  className="chat-room-page"
>
  <div
    className="chat-room-chat-column"
  >
    <div
      className="chat-room-room-info"
    >
      <div
        className="panel panel-primary"
      >
        <div
          className="panel-heading room-info-panel-heading"
        >
          <div
            className="room-info-panel-title"
          >
            <h3
              className="panel-title"
            />
          </div>
          <div
            className="room-info-panel-buttons"
          >
            <button
              className="pull-right btn btn-primary btn-sm"
              onClick={[Function]}
            >
              Edit
            </button>
          </div>
        </div>
      </div>
    </div>
    <ul
      className="chat-message-list list-unstyled"
    />
    <div
      className="chat-room-buttons"
    >
      <div>
        <button
          className="btn btn-primary"
          onClick={[Function]}
        >
          Say something...
        </button>
        <button
          className="btn btn-default pull-right"
          onClick={[Function]}
        >
          Who's here?
        </button>
      </div>
    </div>
  </div>
  <div
    className="chat-room-checklist-column"
  >
    <div
      className="checklist-table-container"
    >
      <table
        className="table checklist-table"
      >
        <thead>
          <tr>
            <th />
            <th />
            <th>
              Start
            </th>
            <th>
              End
            </th>
            <th>
              Description
            </th>
            <th>
              Performed by
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
               
            </td>
            <td>
               
            </td>
            <td
              className="text-success"
              colSpan="4"
            >
              <h4>
                This chat has no checklist items.
              </h4>
              <h4>
                Use the "Add Checklist Item" button below to add one.
              </h4>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="chat-room-buttons"
    >
      <div>
        <button
          className="btn btn-primary checklist-item-button pull-right"
          onClick={[Function]}
        >
          Add Checklist Item...
        </button>
      </div>
    </div>
  </div>
</div>
`;
